.github/workflows/build.yml

```diff
@@ -110,9 +110,12 @@ jobs:
         # --continue:     Keep running even if a test fails
         # -PcommitId      Prevent the Git SHA being written into the jar files (which breaks caching)
         id: junit-test
+        env:
+          TIMEOUT_MINUTES: 180
         run: |
           set +e
-          timeout 180m ./gradlew --build-cache --continue \
+          ./.github/scripts/thread-dump.sh &
+          timeout ${TIMEOUT_MINUTES}m ./gradlew --build-cache --continue \
```

Comment:
That aside, should we add `timeout` to the summary? Maybe we can save the thread dump to a file and then archive it to be able download directly.

Reply:
> That aside, should we add timeout to the summary?

Good idea. Will add

> Maybe we can save the thread dump to a file and then archive it to be able download directly.

This is a great idea :) I'll see what I can do here.